clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -83,6 +85,15 @@ public void configure(Map<String, ?> configs) throws KafkaException {
         if (endpointIdentification != null)
             this.endpointIdentification = endpointIdentification;
 
+        String secureRandomImplementation = (String) configs.get(SslConfigs.SSL_SECURE_RANDOM_IMPLEMENTATION_CONFIG);
+        if (secureRandomImplementation != null) {
+            try {
+                this.secureRandomImplementation = SecureRandom.getInstance(secureRandomImplementation);
+            } catch (GeneralSecurityException e) {
```

Comment:
As a note, that "hypothetical situation" is not uncommon. This happens constantly with security issues.